include/mbedtls/mps/trace.h

```diff
@@ -0,0 +1,113 @@
+
+#ifndef MBEDTLS_MPS_TRACE_H
+#define MBEDTLS_MPS_TRACE_H
+
+/*
+ * Adapt this to enable/disable tracing output
+ * from the various layers of the MPS.
+ */
+
+#define TRACE_ENABLE_LAYER_1
+#define TRACE_ENABLE_LAYER_2
+#define TRACE_ENABLE_LAYER_3
+#define TRACE_ENABLE_READER
+#define TRACE_ENABLE_WRITER
+
+/*
+ * This shouldn't be touched.
+ */
+
+__attribute__((unused)) static int trace_id;
+
+typedef enum
+{
+    trace_comment=0,
+    trace_call,
+    trace_error
+} trace_type;
+
+#define TRACE_ID_LAYER_1 1
```

Comment:
I prefer to keep it this way to be able to see at a glance which bits are used by which sub-module. If I eliminate `TRACE_ID_XXX` by inlining them in the respective `#if defined(TRACE_ENABLE_XXX)` section, this becomes less transparent.

Using `TRACE_BIT_XXX` instead of `TRACE_ID_XXX`.